table/block_based/filter_policy.cc

```diff
@@ -1111,14 +1242,54 @@ class AlwaysTrueFilter : public FilterBitsReader {
  public:
   bool MayMatch(const Slice&) override { return true; }
   using FilterBitsReader::MayMatch;  // inherit overload
+  bool HashMayMatch(const uint64_t) override { return true; }
+  using FilterBitsReader::HashMayMatch;  // inherit overload
+  bool IsAlwaysTrueFilter() override { return true; }
 };
 
 class AlwaysFalseFilter : public FilterBitsReader {
  public:
   bool MayMatch(const Slice&) override { return false; }
   using FilterBitsReader::MayMatch;  // inherit overload
+  bool HashMayMatch(const uint64_t) override { return false; }
+  using FilterBitsReader::HashMayMatch;  // inherit overload
 };
 
+Status XXPH3FilterBitsBuilder::PostVerify(const Slice& filter_content) {
+  Status s = Status::OK();
+
+  if (!detect_filter_construct_corruption_) {
```

Comment:
I haven't read through all the details, but it would seem cleaner to me to make this decision (based on option) before calling PostVerify, so that custom FilterBitsBuilder doesn't have to be involved in that decision.

Reply:
Oh right I forgot about accumulating vs. purging keys, which means we want to know earlier whether we're going to verify.

Reply:
(no need to reply) Discussed offline so we are still taking the MaybePostVerify approach with clarified comment as suggestion below